### PR TITLE
RGR-693 Roger Chat App not supporting attachments anymore

### PR DESCRIPTION
**Pull request by elonmitchell**
Problem
=======
- Roger Chat App not supporting attachments anymore
[link to JIRA RGR-693](https://usxtech.atlassian.net/browse/RGR-693)

Solution
========
- Implemented the preview for the image, video and pdf in s3 link.
- Implemented the s3 bucket link with current time to prevent to replace files for same file name.

Screenshots (optional):
-----------------------
Screenshot of Android device

### CODE REVIEW

File: src/components/Chat.tsx [hunk 1]
```diff
@@ -257,31 +267,82 @@ const Chat = (props: VariantChatProps): ReactElement => {
     getMoreMessages();
   }, [moreMessages]);
 
-  const handlePickDocument = async () => {
+  const handleAttachment = async () => {
     if (!getS3Keys()) {
```
Comment (ajp8164) [inline src/components/Chat.tsx:271]: Can this be removed? It should not be possible to to init chat without proper config (incl these keys). Idk if that’s true but it should be.

File: src/components/Chat.tsx [hunk 2]
```diff
@@ -257,31 +267,82 @@ const Chat = (props: VariantChatProps): ReactElement => {
...
+    if (attachmentActionSheetRef.current) {
+      attachmentActionSheetRef.current.show();
+    }
+  };
 
-    if (!name || !type || !uri) {
-      return;
+  const handleSelectAttachment = (index: number) => {
+    switch (index) {
+      case 0:
```
Comment (ajp8164) [inline src/components/Chat.tsx:283]: Can we use an enumeration here so we don’t use magic numbers ;-)

File: src/components/Chat.tsx [hunk 3]
```diff
@@ -257,31 +267,82 @@ const Chat = (props: VariantChatProps): ReactElement => {
...
-          uri: location,
-        });
+      if (!fileName || !type || !uri) {
+        return;
       }
-    });
+
+      setIsUploading(true);
+
+      uploadOnS3(fileName, type, decodeURI(uri), (location: string | null) => {
```
Comment (ajp8164) [inline src/components/Chat.tsx:309]: I guess there’s a separate ticket for using a folder structure in S3? If it’s super easy to add maybe we can just add now. Idk if it’s as easy as just adding this or not.

driverId + “/“ + unixtimestamp + “/“ ..

File: src/components/Chat.tsx [hunk 4]
```diff
@@ -425,6 +486,13 @@ const Chat = (props: VariantChatProps): ReactElement => {
           ...lightboxProps,
         }}
       />
+      <ActionSheet
+        ref={attachmentActionSheetRef}
+        title=" Add attachment "
+        options={['File', 'Photo / Video', ' Cancel ']}
+        cancelButtonIndex={2}
```
Comment (ajp8164) [inline src/components/Chat.tsx:493]: Use enum for the 2?

**Comment by miked49er** [comment]
Once Andy's comments are addressed I'll approve

File: src/components/Chat.tsx (hunk 1 above)
Comment (elonmitchell) [reply src/components/Chat.tsx:271]: @ajp8164 we need the code for Ops Chat app since we don't have the AWS keys.

**Review by abaidusx — APPROVED** [review]

**Pull request merged**